src/status_im/contexts/wallet/add_account/create_account/view.cljs

```diff
@@ -57,125 +54,216 @@
+                                    [:wallet/derive-address-and-add-account
+                                     {:password             password
+                                      :derived-from-address derived-from
+                                      :derivation-path      @derivation-path
+                                      :account-preferences  preferences}])))
+                               [derived-from])]
+        (rn/use-effect
+         #(rf/dispatch
+           [:wallet/next-derivation-path
+            {:on-success  set-derivation-path
+             :keypair-uid key-uid}])
+         [key-uid])
+
+        [floating-button
+         {:account-color      @account-color
+          :slide-button-props {:on-auth-success on-auth-success
+                               :disabled?       (or (empty? @account-name)
+                                                    (= "" @derivation-path))}}
+         [avatar
+          {:account-color   @account-color
+           :emoji           @emoji
+           :on-select-emoji set-emoji}]
+         [input
+          {:account-color  @account-color
+           :account-name   @account-name
+           :on-change-text on-change-text}]
+         [color-picker
+          {:account-color     @account-color
+           :set-account-color set-account-color}]
+         [new-account-origin
+          {:derivation-path     @derivation-path
+           :customization-color customization-color}]]))))
+
+(defn view
+  [_]
+  (let [account-name      (reagent/atom "")
+        account-color     (reagent/atom (rand-nth colors/account-colors))
+        emoji             (reagent/atom (emoji-picker.utils/random-emoji))
+        on-change-text    #(reset! account-name %)
+        set-account-color #(reset! account-color %)
+        set-emoji         #(reset! emoji %)
+        state             {:account-name  account-name
+                           :account-color account-color
+                           :emoji         emoji}]
+    (fn []
+      (let [customization-color    (rf/sub [:profile/customization-color])
+            ;; Having a keypair means the user is importing it or creating it.
+            {:keys [keypair-name]} (rf/sub [:wallet/create-account-new-keypair])]
+
+        (rn/use-unmount #(rf/dispatch [:wallet/clear-create-account]))
+
+        (if keypair-name
+          [add-new-keypair-variant
+           {:customization-color customization-color
+            :on-change-text      on-change-text
+            :set-account-color   set-account-color
+            :set-emoji           set-emoji
+            :state               state
+            :keypair-name        keypair-name}]
+          [derive-account-variant
+           {:customization-color customization-color
+            :on-change-text      on-change-text
+            :set-account-color   set-account-color
+            :set-emoji           set-emoji
+            :state               state}])))))
```

Comment:
Why did you choose these 2 variants approach? I see considertable duplication in the views

Reply:
Hi @OmarBasem 

I abstracted the views as much as I could so that the amount of code duplicated is small, I could have abstracted one step more to eliminate the duplication, but IMO the code was worse.

The reason is it was mixing two flows:
- Create a derived account for existing keypair
- Create a new keypair an derive a first account

Depending on the flow we need some different things, so, to make this logic clearer, I decided to split it into 2 variants.

Previously, it was hard to understand why we were using some subscriptions if we only want to derive an account, so I hope this change helps the reading/understanding of the functionality.

Reply:
> Previously, it was hard to understand why we were using some subscriptions if we only want to derive an account

Oh I see, makes sense 👍